firebase-functions/src/main/java/com/google/firebase/functions/HttpsCallableOptions.kt

```diff
@@ -0,0 +1,49 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package com.google.firebase.functions
+
+/**
+ * Options for configuring the callable function.
+ *
+ *
+ * These properties are immutable once a callable function reference is instantiated.
+ */
+class HttpsCallableOptions private constructor(
+        /**
+         * Returns the setting indicating if limited-use App Check tokens are enforced for this function.
+         */
+        // If true, request a limited-use token from AppCheck.
+        val limitedUseAppCheckTokens: Boolean) {
+
+  /** Builder class for [com.google.firebase.functions.HttpsCallableOptions]  */
+  class Builder {
+    /** Returns the setting indicating if limited-use App Check tokens are enforced.  */
+    var limitedUseAppCheckTokens = false
+      private set
```

Comment:
I'm probably dreaming too high here, but what if we dropped the Builder completely and just had a simple `HttpsCallableOptions` class that could be passed to `getHttpsCallable()`? It would be a breaking change, but would simplify Kotlin usage, and I believe this change will come out in a release with other breaking changes anyway.

Something like:

```kotlin
class HttpsCallableOptions @JvmOverloads public constructor(
  @JvmField
  val limitedUseAppCheckTokens = false
) {

  init {
    // if the original `build()` method was doing any kind of validation,
    // we could've done it here. But in this particular case, it is not.
  }

}
```

Kotlin would call it like:

```
val options = HttpsCallableOptions(limitedUseAppCheckTokens = true)
```

Java:
```
HttpsCallableOptions options = new HttpsCallableOptions(true);
```

Reply:
> and I believe this change will come out in a release with other breaking changes anyway.

The intention is to avoid this, actually. We wanna [quickly] convert SDKs to Kotlin without breaking anyone, then go back and redesign the APIs as desired. It may make less sense for something as small as functions- but with the larger SDKs it'll make it significantly easier to make progressive changes once everything's in Kotlin. Although, with the new breaking change policy- we may want to revisit that idea.